test/integration/Test/Integration/Scenario/Wallets.hs

```diff
@@ -102,9 +112,504 @@ spec = do
+        let payload = Json [json| {
+                "name": "Secure Wallet",
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": []
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Array"
+            ]
+
+    it "WALLETS_CREATE_07 - Num as passphrase -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": "Secure Wallet",
+                "mnemonic_sentence": #{mnemonics15},
+                "passphrase": 777
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "expected Text, encountered Number"
+            ]
+
+    it "WALLETS_CREATE_07 - passphrase param missing -> fail" $ \ctx -> do
+        let payload = Json [json| {
+                "name": "Secure Wallet",
+                "mnemonic_sentence": #{mnemonics15}
+                } |]
+        r <- request @ApiWallet ctx ("POST", "v2/wallets") Default payload
+        verify r
+            [ expectResponseCode @IO HTTP.status400
+            , expectErrorMessage "key \"passphrase\" not present"
+            ]
+
+    describe "WALLETS_CREATE_08 - address_pool_gap" $ do
+        let matrix =
+                [ ( show addressPoolGapMin, addressPoolGapMin
+                  , [ expectResponseCode @IO HTTP.status202
+                    , expectFieldEqual addressPoolGap addressPoolGapMin
+                    ]
+                  )
+                , ( show (addressPoolGapMin - 1) ++ " -> fail"
+                  , (addressPoolGapMin - 1)
+                  , [ expectResponseCode @IO HTTP.status400
+                    , expectErrorMessage "ErrGapOutOfRange 9"
+                    ] -- probably to be modified after #204
+                  )
+                , ( show addressPoolGapMax, addressPoolGapMax
+                  , [ expectResponseCode @IO HTTP.status202 ]
+                  )
+                , ( show (addressPoolGapMax + 1) ++ " -> fail"
+                  , addressPoolGapMax + 1
+                  , [ expectResponseCode @IO HTTP.status400
+                    , expectErrorMessage "ErrGapOutOfRange 101"
+                    ] -- probably to be modified after #204
+                  )
+                , ( "0 -> fail", 0
+                   , [ expectResponseCode @IO HTTP.status400
+                     , expectErrorMessage "ErrGapOutOfRange 0"
+                     ] -- probably to be modified after #204
+                  )
+                , ( "-1 -> fail", (-1)
+                  , [ expectResponseCode @IO HTTP.status400
+                    , expectErrorMessage "ErrGapOutOfRange 255"
+                    ] -- probably to be modified after #204
```

Comment:
@KtorZ - I pushed tests for `address_pool_gap` here after all. Also created #204 as discussed.